pkg/ebpf/c/common/buffer.h

```diff
@@ -376,8 +388,9 @@ statfunc int save_str_arr_to_buf(args_buffer_t *buf, const char __user *const __
 statfunc int save_args_str_arr_to_buf(
     args_buffer_t *buf, const char *start, const char *end, int elem_num, u8 index)
 {
-    // Data saved to submit buf: [index][len][arg_len][arg #][null delimited string array]
-    // Note: This helper saves null (0x00) delimited string array into buf
+    // Data saved to submit buf: [index][len][arg_len][arg #][array of null delimited string]
+    // Note: This helper saves null (0x00) delimited string array into buf in the format:
+    // [[str1][\n][str2][\n][...][strn][\n])]
```

Comment:
This comment is not so clear to me.

First doubt: Is the delimiter of this array elements a single NUL char or a NULL pointer?
Second doubt: Is the final delimiter of the array the same size of the elements delimiter?

Reply:
Each string is size variable. The indication for the end of a string is just a null terminator char. You can know which \n is the final one using the arg_len provided beforehands.

Reply:
So would it be like this?

`NUL = '\0'`

`[ char_arr1[n], NUL, char_arr2[n], NUL, ... char_arrn[n], NUL ]`

If so, consider it as a suggestion.

Another way, even more expressive might be:

```
    [ char_arr1[0]  char_arr1[1]  ...  char_arr1[n-1]  '\0',
      char_arr2[0]  char_arr2[1]  ...  char_arr2[n-1]  '\0',
      ...
      char_arrn[0]  char_arrn[1]  ...  char_arrn[n-1]  '\0' ]
```